test_data/script/validate.py

```diff
@@ -0,0 +1,107 @@
+# This code validates input files ("study_test.tsv", "participant_test.tsv", "specimen_test.tsv" and "sample_test.tsv")
+# against JSON schema ("dictionary.json"). 
+# It 1) validates that "required" fields are present; and
+#    2) validates data types and the match to defined lists and specific regular expressions. 
+# Author: Guanqiao Feng
+# Date: 2024-04-02
+
+import pandas as pd
+import json
+import re
+import sys
+import glob
+import os
+
+# Function to validate required columns for a given schema
+def validate_required_columns(df, schema, schema_name):
+    required_fields = [field['name'] for field in schema['fields'] if field['restrictions'].get('required', False)]
+    cols_ok = all(col in df.columns for col in required_fields)
+    print(f"{schema_name}: All required columns present -> {cols_ok}")
+
+# Mapping between JSON schema types and pandas types
+type_mapping = {
+    'string': 'object',  # In pandas, strings are typically stored as objects
+    # Add more mappings as needed, e.g., 'integer': 'int64', 'number': 'float64', etc.
+}
+
+# Function to check if data matches the defined list (enumeration)
+def validate_code_list(column, code_list):
+    invalid_rows = []
```

Comment:
Row numbers are included in the error message. Addressed https://github.com/Pan-Canadian-Genome-Library/data-dictionary/pull/3#discussion_r1556364008